### PR TITLE
Clarified language about what external drivers are in the context of …

### PR DESCRIPTION
**Pull request by xoedusk**
## Description

- Clarified language about what *external drivers* are in the context of LEDs. It was not clear what they were 'external' to. 
- Added language that the RGB Matrix system is commonly used for per-key backlighting.
- Replaced **LED's** in nearby sentence with **LEDs** to make it consistent with much of the rest of the document. Plural, not possessive.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_led_matrix.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # LED Matrix Lighting :id=led-matrix-lighting
 
-This feature allows you to use LED matrices driven by external drivers. It hooks into the backlight system so you can use the same keycodes as backlighting to control it.
+This feature allows you to use non-RGB LED matrices driven by external drivers. *External* here means *not within the microcontroller,* and includes LEDs with embedded drivers. LED Matrix Lighting hooks into the backlight system so you can use the same keycodes as backlighting to control it.
```
Comment (fauxpark) [inline docs/feature_led_matrix.md:3]:
```suggestion
This feature allows you to use LED matrices controlled by external LED driver ICs.
```

File: docs/feature_rgb_matrix.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # RGB Matrix Lighting :id=rgb-matrix-lighting
 
-This feature allows you to use RGB LED matrices driven by external drivers. It hooks into the RGBLIGHT system so you can use the same keycodes as RGBLIGHT to control it.
+This feature allows you to use RGB LED matrices driven by external drivers. *External* here means *not within the microcontroller,* and includes LEDs with embedded drivers. RGB Matrix Lighting is commonly used for per-key backlighting. It hooks into the RGBLIGHT system so you can use the same keycodes as RGBLIGHT to control it.
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md:3]:
```suggestion
This feature allows you to use RGB LED matrices controlled by external LED driver ICs.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Way too wordy, can be reduced down to this.
These entire pages will likely be completely rewritten soon anyway...

**Comment by xoedusk** [comment]
On an initial reading of the original document by a non-veteran (me), I had to look up what an 'external driver' was. It did not help. On the qmk discord server, it was suggested that I google the first external driver example given on the RGB Matrix page, which was IS31FL3731.  This led to looking at a completely separate chip like https://www.adafruit.com/product/2946

Thus, the top-most information on the page isn't clear that this includes per-key LEDs that have drivers embedded within them. My change attempts to address this issue. I don't believe @fauxpark 's suggested modification of my change addresses the issue.

**Comment by xoedusk** [comment]
This perhaps clarifies the confusion I am trying to address.
Screenshot 2023-12-09 at 9 48 48 AM

**Comment by fauxpark** [comment]
> Am I the only one that interprets "external drivers" as not on the LED?

Yes. You are. Because QMK does not run on the LEDS.

**Comment by fauxpark** [comment]
I should also note, I am currently in the middle of cleaning up the LED/RGB Matrix driver code, which also means I will be moving that documentation out of here and creating separate pages for each one. As I said, that will probably have the side effect of rewriting parts of the LED/RGB Matrix docs pages to clarify your issue anyway.

**Comment by xoedusk** [comment]
Closed pull request for the above reasons.
